Conceptpower+Spring/src/main/java/edu/asu/conceptpower/root/DatabaseClient.java

```diff
@@ -23,336 +24,337 @@
+     * @see edu.asu.conceptpower.db4o.IConceptDBManager#store(java.lang.Object, java.lang.String)
+     */
+    @Override
+    public void store(Object element, String databasename) {
+        // FIXME no caching?
+        // if (databasename.equals(DBNames.WORDNET_CACHE)) {
+        // wordnetCacheClient.store(element);
+        // wordnetCacheClient.commit();
+        // return;
+        // }
+        if (databasename.equals(DBNames.DICTIONARY_DB)) {
+            dictionaryClient.store(element);
+            dictionaryClient.commit();
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.db4o.IConceptDBManager#update(edu.asu.conceptpower.core.ConceptEntry, java.lang.String)
+     */
+    @Override
+    public void update(ConceptEntry entry, String databasename) {
+        if (databasename.equals(DBNames.DICTIONARY_DB)) {
+            ConceptEntry toBeUpdated = getEntry(entry.getId());
+            toBeUpdated.setBroadens(entry.getBroadens());
+            toBeUpdated.setConceptList(entry.getConceptList());
+            toBeUpdated.setDescription(entry.getDescription());
+            toBeUpdated.setEqualTo(entry.getEqualTo());
+            toBeUpdated.setNarrows(entry.getNarrows());
+            toBeUpdated.setPos(entry.getPos());
+            toBeUpdated.setSimilarTo(entry.getSimilarTo());
+            toBeUpdated.setSynonymIds(entry.getSynonymIds());
+            toBeUpdated.setSynsetIds(entry.getSynsetIds());
+            toBeUpdated.setTypeId(entry.getTypeId());
+            toBeUpdated.setWord(entry.getWord());
+            toBeUpdated.setWordnetId(entry.getWordnetId());
+            toBeUpdated.setDeleted(entry.isDeleted());
+            toBeUpdated.setChangeEvents(entry.getChangeEvents());
+            dictionaryClient.store(toBeUpdated);
+            dictionaryClient.commit();
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.db4o.IConceptDBManager#deleteConceptList(java.lang.String)
+     */
+    @Override
+    public void deleteConceptList(String name) {
+        ConceptList list = new ConceptList();
+        list.setConceptListName(name);
+
+        ObjectSet<ConceptList> results = dictionaryClient.queryByExample(list);
+        for (ConceptList res : results) {
+            dictionaryClient.delete(res);
+            dictionaryClient.commit();
+        }
+    }
+
+    /* (non-Javadoc)
+     * @see edu.asu.conceptpower.db4o.IConceptDBManager#update(edu.asu.conceptpower.core.ConceptList, java.lang.String, java.lang.String)
+     */
+    @Override
+    public void update(ConceptList list, String listname, String databasename) {
+        if (databasename.equals(DBNames.DICTIONARY_DB)) {
+            ConceptList toBeUpdated = getConceptList(listname);
+            toBeUpdated.setConceptListName(list.getConceptListName());
+            toBeUpdated.setDescription(list.getDescription());
+            dictionaryClient.store(list);
+            dictionaryClient.commit();
+        }
+    }
+
```

Comment:
@jdamerow  I changed in this method: 	public void update(ConceptEntry entry, String databasename).  I added one more line to include changeEvents list and removed modified name.